### DISCUSSION TITLE
Loading (spinner) for ModalContent or Card?

### DISCUSSION THREAD

**Discussion by sturlath** [Megabit/Blazorise#2429] [Q&A] [answered]
I'm opening a Modal that creates Azure data that I then display in the model. Now I just show the Modal and wait until the data shows up. I would like to convey to the user that something is going on.

How can I have all the modal window "spinning" or show something spinning over the fields (or whole/part of a Card) that will hold the data? 

Button has Loading, is it possible to extend that to all elements?

**Comment by stsrki** [comment, marked as answer]
Maybe a better and less intrusive option is to use the PageProgress? 

docs: https://blazorise.com/docs/components/progress/#page-progress
demo: https://bootstrapdemo.blazorise.com/tests/progressbars

Nothing else comes to mind currently in v0.9.3.

Recently I have implemented Blazorise.SpinKit extension in v0.9.4 and would be perfect for this scenario.

**Reply by sturlath** [reply]
Awesome! I think  progress at the top of the page might solve this. Will try out this SpinKit when that comes out :-)

**Reply by sturlath** [reply]
Ok I have tried to put this line `<PageProgress Visible="true" />` just about anywhere I could think of in my code without any success of it showing up.  Some trick to this or is it broken?

I´m using abp.io 4.3.2 and Blazorize 0.9.3.5

**Reply by stsrki** [reply]
There should not be any trick. Must be something you're missing. Have a look at the demo code https://github.com/Megabit/Blazorise/blob/master/Demos/Blazorise.Demo/Pages/Tests/ProgressBarsPage.razor

**Reply by sturlath** [reply]
humm... I copied this code below and nothing shows up... 

```csharp
PageProgress pageProgress;
Color pageProgressColor = Color.None;
bool pageProgressVisible = false;

private async Task OnShowPageProgressIndeterminate()
{
  pageProgressColor = Color.None; //Tried also Danger...
  pageProgressVisible = true;

  await Task.Delay(TimeSpan.FromSeconds(4));

  pageProgressVisible = false;
}
```   

and then I added a button at the top of the page

```xml
 <Button Color="Color.Primary" Clicked="@OnShowPageProgressIndeterminate">Indeterminate progress</Button>
```

**Reply by stsrki** [reply]
Really strange. The same component is implemented in ABP. In fact, I have integrated it a few months ago and was working just fine back then.

**Reply by sturlath** [reply]
Yes I dont know what to try out 🙈 Ill propably just wait for 9.4 and SpinKit 😉 lot to do before launch..

**Reply by stsrki** [reply]
Too much to do 😅
